simphony/cuds/tests/test_mesh.py

```diff
@@ -680,5 +682,48 @@ def test_update_cell(self):
 
         self.assertItemsEqual(cell_upd.points, cell_ret.points)
 
+    def test_set_data(self):
+        """ Check that data can be retrieved
+
+        """
+
+        org_data = DataContainer()
+
+        org_data[CUBA.VELOCITY] = (0, 0, 0)
+
+        self.mesh.data = org_data
+        ret_data = self.mesh.data
+
+        self.assertItemsEqual(org_data, ret_data)
```

Comment:
We also need to check that 
```python
self.assertIsNot(org_data, ret_data)
```

So that we verify that they are equal but not the same instance as described in #83